### PR TITLE
Make sure machine render happen successfully after pull secret removal

### PR DESCRIPTION
**Pull request by praveenkumar**
With this patch we make sure that after the removing the pull secret
machine config and machine config pool successfully rendered and not
in inconsistent state. This is causing issue on the CI where render
didn't happen correctly.

### CODE REVIEW

File: pkg/crc/cluster/cluster.go [hunk 1]
```diff
@@ -179,16 +179,43 @@ func EnsureGeneratedClientCAPresentInTheCluster(ocConfig oc.Config, sshRunner *s
...
+		mcAfterRemovePullSecret, stderr, err := ocConfig.RunOcCommand("get mc --sort-by=.metadata.creationTimestamp --no-headers -oname")
+		if err != nil {
+			return &errors.RetriableError{Err: fmt.Errorf("failed to get machineconfig resource %w: %s", err, stderr)}
+		}
+		if mcBeforeRemovePullSecret == mcAfterRemovePullSecret {
+			return &errors.RetriableError{Err: fmt.Errorf("machine config is not rentered after removing pull secret")}
+		}
+		// This 20sec sleep is required because just after machine config render happen it takes some time
+		// for machine config pool to process this new render file.
+		time.Sleep(20 * time.Second)
```
Comment (guillaumerose) [inline pkg/crc/cluster/cluster.go:206]: Maybe split in 2 waiting loops ?


Something like:

```
        // wait for new mc
        mcRenderedFunc := func() error {
		mcAfterRemovePullSecret, stderr, err := ocConfig.RunOcCommand("get mc --sort-by=.metadata.creationTimestamp --no-headers -oname")
		if err != nil {
			return &errors.RetriableError{Err: fmt.Errorf("failed to get machineconfig resource %w: %s", err, stderr)}
		}
		if mcBeforeRemovePullSecret == mcAfterRemovePullSecret {
			return &errors.RetriableError{Err: fmt.Errorf("machine config is not rentered after removing pull secret")}
		}
	}
	if err := errors.RetryAfter(1*time.Minute, mcRenderedFunc, 2*time.Second); err != nil {
		return err
	}

        // wait for mc to be used and deployed
        mcRenderedFunc := func() error {
		mcpDegradedMachineCount, stderr, err := ocConfig.RunOcCommand("get mcp master -ojsonpath='{.status.degradedMachineCount}'")
		if err != nil {
			return &errors.RetriableError{Err: fmt.Errorf("failed to get machineconfig resource %w: %s", err, stderr)}
		}
		if strings.TrimSpace(mcpDegradedMachineCount) != "0" {
			return &errors.RetriableError{Err: fmt.Errorf("machine pool config is in degraded machine count")}
		}
		return nil
	}
	if err := errors.RetryAfter(1*time.Minute, mcRenderedFunc, 2*time.Second); err != nil {
		return err
	}

```

Comment (praveenkumar) [reply pkg/crc/cluster/cluster.go:206]: @guillaumerose here the first loop finish as soon as the master-render appears in `machine-config` and when I test this, `machine-config-pool (mcp)` doesn't start progressing asap, it does take some time, I really don't want any sleep time but don't have any better way to make sure it is handled properly.

Comment (guillaumerose) [reply pkg/crc/cluster/cluster.go:206]: ok let's go with this

File: pkg/crc/cluster/cluster.go [hunk 2]
```diff
@@ -179,16 +179,43 @@ func EnsureGeneratedClientCAPresentInTheCluster(ocConfig oc.Config, sshRunner *s
...
 		return fmt.Errorf("Failed to remove Pull secret %w: %s", err, stderr)
 	}
 
+	mcRenderedFunc := func() error {
+		mcAfterRemovePullSecret, stderr, err := ocConfig.RunOcCommand("get mc --sort-by=.metadata.creationTimestamp --no-headers -oname")
+		if err != nil {
+			return &errors.RetriableError{Err: fmt.Errorf("failed to get machineconfig resource %w: %s", err, stderr)}
+		}
+		if mcBeforeRemovePullSecret == mcAfterRemovePullSecret {
+			return &errors.RetriableError{Err: fmt.Errorf("machine config is not rentered after removing pull secret")}
```
Comment (anjannath) [inline pkg/crc/cluster/cluster.go:202]: small typo `rentered` should be `rendered`

Comment (praveenkumar) [reply pkg/crc/cluster/cluster.go]: Fixed

**Pull request merged**